sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -170,4 +174,27 @@ abstract class QueryPlan[PlanType <: TreeNode[PlanType]] extends TreeNode[PlanTy
   protected def statePrefix = if (missingInput.nonEmpty && children.nonEmpty) "!" else ""
 
   override def simpleString: String = statePrefix + super.simpleString
+
+  protected def childrenJson: JValue = {
+    if (children.isEmpty) {
+      JNothing
+    } else if (children.length == 1) {
+      "child" -> children.head.jsonValue
```

Comment:
i think this actually makes parsing harder -- maybe just always have "children"?